code/mob/living.dm

```diff
@@ -795,6 +798,15 @@
 						secure_headset_mode = lowertext(copytext(message,2,3))
 					message = copytext(message, 3)
 
+	// check for singing prefix
+	if (!singing && dd_hasprefix(message, singing_prefix) && !src.stat)
+		singing = 1
+		message = copytext(message, 2)
+		if (src.bioHolder.HasEffect("accent_scots"))
+			danny_index++
+			if (danny_index > 16)
+				danny_index = 1
```

Comment:
are you trying to make it so that they 'danny' every 15 times they try to sing? That's what this looks like it does.

Reply:
also this var does not appear to be used

Reply:
each time a Scottsman sings they sing the next part of Danny Boy. danny_index keeps track of which line they are up to